misc-scripts/xref_mapping/XrefParser/EntrezGeneParser.pm

```diff
@@ -50,74 +50,29 @@ sub run {
   
 
   my $eg_io = $self->get_filehandle($file);
-  if ( !defined $eg_io ) {
-    print STDERR "ERROR: Could not open $file\n";
-    return 1;    # 1 is an error
-  }
+  croak "ERROR: Could not open $file\n" unless defined $eg_io;
 
-  my %seen;
+  my $input_file = Text::CSV->new({ sep_char           => "\t",
+				    empty_is_undef     => 1,
```

Comment:
Could you turn on strict here instead of checking for errors later?

Reply:
From the docs: "If this attribute (strict) is set to 1, any row that parses to a different number of fields than the previous row will cause the parser to throw error 2014."

I am not checking for errors later, other than checking parsing has reached EOF. This does not seem to apply to the above.

Reply:
Or maybe you meant that for the HPA parser?!